clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5495,11 +5495,7 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
       }
     } else {
       // Use the default.
-      llvm::Triple TT(Triple);
-      TT.setArch(llvm::Triple::spir64);
-      TT.setVendor(llvm::Triple::UnknownVendor);
-      TT.setOS(llvm::Triple(llvm::sys::getProcessTriple()).getOS());
-      TT.setEnvironment(llvm::Triple::SYCLDevice);
+      llvm::Triple TT(C.getDriver().MakeSYCLDeviceTriple());
       TargetInfo += TT.normalize();
```

Comment:
Same as other comment, doesn't look like TT is really needed and just append the MakeTriple